internal/exec/copy_glob_test.go

```
@@ -0,0 +1,663 @@
+// Use a local variable to override the glob matching function in tests.
+var getGlobMatchesForTest = utils.GetGlobMatches
+
+// Helper that calls our local getGlobMatchesForTest.
+func getMatchesForPatternForTest(sourceDir, pattern string) ([]string, error) {
+	fullPattern := filepath.Join(sourceDir, pattern)
+	// Normalize fullPattern to use forward slashes.
+	normalized := filepath.ToSlash(fullPattern)
+	return getGlobMatchesForTest(normalized)
+}
+
+type fakeDirEntry struct {
```

golangci-lint (1.62.2)

type `fakeDirEntry` is unused

(unused)